### PR TITLE
pydantic crunchy model

### PR DESCRIPTION
**Pull request by moonso**
## Description
This PR adds so that the crunchy api makes use of the metadata models described in https://github.com/Clinical-Genomics/cgmodels/blob/main/cgmodels/crunchy/metadata.py

This PR introduces a dependency on `cgmodels`

### How to test
Tests are automated

### Expected test outcome
- [x] check that tests pass
- [x] Take a screenshot and attach or copy/paste the output.
Screen Shot 2021-03-11 at 11 18 14

## Review
- [x] code approved by PG
- [x] tests executed by @GithubActions
- [x] "Merge and deploy" approved by MM
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [x] **MINOR** - when you add functionality in a backwards compatible manner

### CODE REVIEW

File: cg/apps/crunchy/crunchy.py [hunk 1]
```diff
@@ -137,24 +135,24 @@ def is_fastq_compression_done(self, compression_obj: CompressionData) -> bool:
             return False
         LOG.info("SPRING metadata file found")
 
-        spring_metadata = self.get_spring_metadata(compression_obj.spring_metadata_path)
+        spring_metadata: CrunchyMetadata = get_spring_metadata(compression_obj.spring_metadata_path)
```
Comment (patrikgrenfeldt) [inline cg/apps/crunchy/crunchy.py:138]: Why have "spring" metadata on one side and "Crunchy" metadata on the other?

File: cg/apps/crunchy/crunchy.py [hunk 2]
```diff
@@ -172,16 +170,17 @@ def is_spring_decompression_done(self, compression_obj: CompressionData) -> bool
             LOG.info("No SPRING metadata file found")
             return False
 
-        spring_metadata = self.get_spring_metadata(spring_metadata_path)
-        if not spring_metadata:
+        try:
+            spring_metadata: CrunchyMetadata = get_spring_metadata(spring_metadata_path)
```
Comment (patrikgrenfeldt) [inline cg/apps/crunchy/crunchy.py:174]: same here

File: cg/apps/crunchy/crunchy.py [hunk 3]
```diff
@@ -214,19 +213,21 @@ def spring_to_fastq(self, compression_obj: CompressionData, sample_id: str = "")
         Decompress SPRING into FASTQ by submitting sbatch script to SLURM
 
         """
-        spring_metadata = self.get_spring_metadata(compression_obj.spring_metadata_path)
+        crunchy_metadata: CrunchyMetadata = get_spring_metadata(
```
Comment (patrikgrenfeldt) [inline cg/apps/crunchy/crunchy.py:216]: same

File: cg/apps/crunchy/files.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+from typing import Dict, List, Optional
+
+from cg.utils.date import get_date
+from cgmodels.crunchy.metadata import CrunchyFile, CrunchyMetadata
+from pydantic import ValidationError
+
+LOG = logging.getLogger(__name__)
+
+
+def get_spring_metadata(metadata_path: Path) -> CrunchyMetadata:
```
Comment (patrikgrenfeldt) [inline cg/apps/crunchy/files.py:15]: same

File: cg/utils/date.py [hunk 5]
```diff
@@ -64,14 +65,14 @@ def get_date(date, date_format=None):
             LOG.info("Date is not in std format")
 
     for separator in [DASH, SPACE, DOT, FWD_SLASH]:
-        splited_date = date.split(separator)
-        if len(splited_date) == 3:
-            return datetime.datetime(*(int(number) for number in splited_date))
+        split_date = date.split(separator)
```
Comment (patrikgrenfeldt) [inline cg/utils/date.py:68]: suggests naming it data_parts

File: tests/apps/crunchy/conftest.py [hunk 6]
```diff
@@ -68,7 +41,12 @@ def fixture_spring_metadata(compression_object: CompressionData) -> List[dict]:
         },
         {"path": str(spring_path), "file": "spring"},
     ]
-    return metadata
+
+
+@pytest.fixture(name="spring_metadata_object")
+def fixture_spring_metadata_object(spring_metadata: List[dict]) -> CrunchyMetadata:
```
Comment (patrikgrenfeldt) [inline tests/apps/crunchy/conftest.py:47]: same

**Review by patrikgrenfeldt — APPROVED** [review]
👍 Great! I think it is a bit confusing that "spring" and "crunchy" are seemingly used interchangeably, then we could use just one name?

File: cg/apps/crunchy/crunchy.py (hunk 1 above)
Comment (moonso) [reply cg/apps/crunchy/crunchy.py:138]: fixed!

File: cg/apps/crunchy/crunchy.py (hunk 2 above)
Comment (moonso) [reply cg/apps/crunchy/crunchy.py:174]: Fixed!

File: cg/apps/crunchy/crunchy.py (hunk 3 above)
Comment (moonso) [reply cg/apps/crunchy/crunchy.py:216]: Fixed!

File: cg/apps/crunchy/files.py (hunk 4 above)
Comment (moonso) [reply cg/apps/crunchy/files.py:15]: Fixed!

File: cg/utils/date.py (hunk 5 above)
Comment (moonso) [reply cg/utils/date.py:68]: Fixed!

File: tests/apps/crunchy/conftest.py (hunk 6 above)
Comment (moonso) [reply tests/apps/crunchy/conftest.py]: Fixed!

**Comment by moonso** [comment]
Screen Shot 2021-03-11 at 11 29 40

Deployed on hasta

**Pull request merged**